src/ec/packages/ec-component-page-banner/page-banner.html.twig

```diff
@@ -0,0 +1,67 @@
+{% spaceless %}
+
+{#
+  Parameters:
+  - "type" (string) (default: 'default') Type of banner (can be 'default','image','image-shade','primary')
+  - "title" (string) (default: '') Title of banner
+  - "image" (string) (default: '') Image for banner (required for image banner type)
+  - "baseline" (string) (default: '') Baseline of banner
+  - "centered" (bool) (default: true) Define if banner should be centered
+  - "button" (associative array) (default: predefined structure) predefined structure for EC Button component
+  - "extra_classes" (optional) (string) (default: '')
+  - "extra_attributes" (optional) (array) (default: [])
+    - "name" (string) Attribute name, eg. 'data-test'
+    - "value" (string) Attribute value, eg: 'data-test-1'
+#}
+
+{# variables #}
+
+{% set _css_class = 'ecl-page-banner' %}
+{% set _extra_attributes = '' %}
+{% set _type = type|default('default') %}
+{% set _centered = centered|default(true) %}
+{% set _title = title|default('') %}
+{% set _baseline = baseline|default('') %}
+{% set _image = image|default('') %}
+{% set _button = button|default({}) %}
+
+{# Internal logic - Process properties #}
+
+{% set _css_class = _css_class ~ ' ecl-page-banner--' ~ _type %}
+
+{% if _centered == true %}
+  {% set _css_class = _css_class ~ ' ecl-page-banner--centered' %}
+{% endif %}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<section class="{{ _css_class }}"{{ _extra_attributes }}>
+  {% if _image is not empty and _type in ['image','image-shade'] %}
+    <div class="ecl-page-banner__image" style="background-image:url({{ _image }})"></div>
+  {% endif %}
+  <div class="ecl-container ecl-page-banner__container">
+    <div class="ecl-page-banner__content">
+      <h1 class="ecl-page-banner__title">{{ _title }}</h1>
+      {% if _baseline is not empty %}
+        <p class="ecl-page-banner__baseline">{{ _baseline }}</p>
+      {% endif %}
+      {% if _button is not empty %}
+        {%- include '../ec-component-button/button.html.twig' with _button|merge({ 
```

Comment:
dashes are not needed here

Reply:
I guess I will never understand when we have to put dashes or not...

Reply:
here you don't need them because the `include` will print a new HTML tag, and is surrounded by either `</p>` or `</h1>` before, and `</div>` after -> `spaceless` will remove the whitespaces between the HTML tags -> 🎉